### PR TITLE
BCC 4 - Architecture initial and updates. Deploys automagically on AWS

### PR DESCRIPTION
**Pull request by KeeganOReilly**
Commits show history.

### CODE REVIEW

**Review by GerniVisser-bbd — DISMISSED** [review]

**Review by Thato-Tladi — DISMISSED** [review]

**Review by GerniVisser-bbd — APPROVED** [review]

File: trader-g-api/target/classes/com/traderg/tradergapi/TraderGApiApplication.class
Comment (GerniVisser-bbd) [inline trader-g-api/target/classes/com/traderg/tradergapi/TraderGApiApplication.class:1]: Remove these please

File: trader-g-api/target/test-classes/com/traderg/tradergapi/TraderGApiApplicationTests.class
Comment (GerniVisser-bbd) [inline trader-g-api/target/test-classes/com/traderg/tradergapi/TraderGApiApplicationTests.class:1]: Aswell as this

File: pom.xml
Comment (GerniVisser-bbd) [inline pom.xml:1]: pom has to be in beancards. not sure what impact this would have on your other stuff

File: cli/pom.xml [hunk 1]
```diff
@@ -15,4 +15,9 @@
       <scope>test</scope>
     </dependency>
   </dependencies>
+
+  <properties>
+    <maven.compiler.source>17</maven.compiler.source>
```
Comment (GerniVisser-bbd) [inline cli/pom.xml]: We should stick to one version of java and I believe Rudolf said to use Java 21

**Review by GerniVisser-bbd — CHANGES_REQUESTED** [review]

**Review by GerniVisser-bbd — APPROVED** [review]

**Review by Thato-Tladi — APPROVED** [review]

**Pull request merged**